swagger_server/__main__.py

```diff
@@ -1,53 +1,31 @@
 #!/usr/bin/env python3
 import os
-from random import randrange
 
 import connexion
-from flask import g
-from sqlalchemy import create_engine
-from sqlalchemy.orm import scoped_session, sessionmaker
+from neomodel import config, db
 
 from swagger_server import encoder
-from swagger_server.orm.DistanceORM import DistanceORM
+from swagger_server.orm.DistanceORM import SampleNode
 
 
-def connect_db():
-    engine = create_engine('mysql://{user}:{passwd}@localhost/{db}'.format(
-        user=os.environ.get('MYSQL_USER'),
-        passwd=os.environ.get('MYSQL_PASSWORD'),
-        db=os.environ.get('MYSQL_DATABASE')
-    ), convert_unicode=True)
-    return scoped_session(sessionmaker(autocommit=False, autoflush=False, bind=engine)), engine
-
-
-def init_db():
-    db_session, engine = connect_db()
-    DistanceORM.query = db_session.query_property()
-    DistanceORM.metadata.create_all(bind=engine)
-
-    with open('swagger_server/test/data/sample.list') as f:
-        test_data = f.read().splitlines()
-        distances = []
-        for line in test_data:
-            s1 = line.rstrip()
-            for s2 in test_data:
-                s2 = s2.rstrip()
-                if s2 != s1:
-                    d = randrange(9)
-                    distances.append(DistanceORM(s1=s1, s2=s2, d=d))
-        db_session.bulk_save_objects(distances)
-        db_session.commit()
-
-
-def get_db():
-    if 'db' not in g:
-        g.db, _ = connect_db()
-
-    return g.db
+@db.transaction
+def create_dummy_data():
+    samples = []
+    for name in ['s1', 's2', 's3']:
+        try:
+            s = SampleNode.nodes.get(name=name)
+        except SampleNode.DoesNotExist:
+            s = SampleNode(name=name).save()
+        samples.append(s)
+    samples[0].neighbors.connect(samples[1], {'dist': 5})
+    samples[0].neighbors.connect(samples[2], {'dist': 10})
```

Comment:
Does this mean `s1` have 2 neighbours `s2` and `s3`? But when I searched for 's1', I got only 's2' in the result.

Reply:
You mean using Python (`SampleNode.get(name='s1').neighbors`)?

Reply:
No, I mean when I calling the API served by the docker container (with this dummy data).

Reply:
`C02W11SKHV2R:paste_vcf zl$ curl --url http://192.168.99.101:8080/distance   --header 'content-type: application/json'   --data '{
"experiment_id": "s1"
}'
{
  "result": [
    {
      "name": "s2"
    }
  ],
  "subType": "nearest-neighbor",
  "type": "distance"
}
`

Reply:
Ah, the prototype API returns the neighbours that's has distance < 10: https://github.com/iqbal-lab-org/dist-api-prototype/pull/2/files#diff-62857f1e2c6c9a1955e06fd24eddd186R22